test/testunusedvar.cpp

```diff
@@ -407,6 +407,29 @@ class TestUnusedVar : public TestFixture {
             "   F f;\n"
             "}");
         ASSERT_EQUALS("", errout.str());
+
+        functionVariableUsage(
+            "int sideEffectFunc();\n"
+            "class F {\n"
+            "public:\n"
+            "   F() : x(sideEffectFunc()) {}\n"
+            "   int x;\n"
+            "};\n"
+            "void f() {\n"
+            "   F f;\n"
+            "}");
+        ASSERT_EQUALS("", errout.str());
+
+        functionVariableUsage(
+            "class F {\n"
+            "public:\n"
+            "   F() : x(unknownFunc()) {}\n"
```

Comment:
in theory we should also be careful about assignments and increments/decrements..

     struct Fred1 {
          Fred1() : x(++someVar) {}
      };
     struct Fred2 {
          Fred2() : x(someVar+=3) {}
      };

It seems to be weird code but who knows..

Reply:
Current implementation takes care about such cases. I've added few more tests.

Reply:
ok!